src/main/java/com/walmartlabs/x12/dex/dx894/DefaultDex894Validator.java

```diff
@@ -65,7 +65,10 @@ public DefaultDex894Validator(CyclicRedundancyCheck crc16) {
     protected X12ErrorDetail compareTransactionCounts(Dex894 dex) {
         X12ErrorDetail detail = null;
         int actualTransactionCount = (dex.getTransactions() != null ? dex.getTransactions().size() : 0);
-        if (dex.getNumberOfTransactions() != actualTransactionCount) {
+        if (dex.getNumberOfTransactions() == null) {
+            detail = new X12ErrorDetail(DefaultDex894Parser.APPLICATION_TRAILER_ID, "DXE02",
+                "DXE02(transaction count) can not be null.");
```

Comment:
Not sure what the current text is but something like 
"Invalid document DXE02 is missing"